### PR TITLE
Copter: use prearm check routine from AP_Arming

### PR DESCRIPTION
**Pull request by peterbarker**
Incorporates / is based on https://github.com/ArduPilot/ardupilot/pull/6888

### CODE REVIEW

**Comment by amilcarlucas** [comment]
yet another successful patch 👍 
You remove more lines than you add :)

**Comment by peterbarker** [comment]
@amilcarlucas Yep, that's always a happy thing.  There's more to cut, but it will involve functionality change.  So, for example, in prearms we honour the bit in the safety mask, but in the arming checks we don't - and that's nuts.

File: ArduSub/radio.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ void Sub::init_rc_out()
     motors.set_throttle_range(channel_throttle->get_radio_min(), channel_throttle->get_radio_max());
 
     // enable output to motors
-    if (arming.rc_check()) {
+    if (arming.rc_calibration_checks(true)) {
```
Comment (OXINARF) [inline ArduSub/radio.cpp]: Doesn't need the `true` like it didn't before 🙂

**Comment by peterbarker** [comment]
On Wed, 6 Sep 2017, Francisco Ferreira wrote:

> I'll probably want to squash many of those Copter commits in the merge if
> you won't mind.

I was expecting lots of squashing to happen there.

> -    if (arming.rc_check()) {
> +    if (arming.rc_calibration_checks(true)) {
> 
> Doesn't need the true like it didn't before 🙂

Hmmm.... ugh.  I don't like this being different to the other vehicles 
considering it is now an @override.  I wonder if @jaxxzer would mind if we 
removed the defaulting.

**Comment by jaxxzer** [comment]
>  I wonder if @jaxxzer would mind if we removed the defaulting.

@peterbarker I wouldn't.

**Comment by jaxxzer** [comment]
LGTM

**Comment by amilcarlucas** [comment]
ping

**Comment by amilcarlucas** [comment]
It works fine, approved

**Comment by peterbarker** [comment]
> I would like it not to give the sad tone right after booting.
> It should only do a sad tone if it already did a "happy tone" :)

Done!

However:

> You remove more lines than you add :)

This added lines!

**Comment by rmackay9** [comment]
Merged, thanks!

Note that there are a few changes in behaviour:

- the pre-arm will not pass until the safety switch is pressed.   Previously copter's arming LED would go green regardless of the arming switch.
- the pre-arm now checks compass and battery.. I think.  Maybe previously it would run the arming checks as part of the pre-arm so these were covered.. I'm not totally sure.

One existing thing that I really don't like about the AP_Arming library is it's use of the AP_Notify::flags.failsafe within the battery_checks and manual_transmitter_checks.  I've said this before and it's unrelated to this PR but the AP_Notify is not meant to be used as a bus for transporting info around the system.  It's just a way to get messages out to the user.  As a test, it should be possible to remove AP_Notify and there should be no change in functionality beyond the lack of LEDs and other signals to the user.

**Comment by amilcarlucas** [comment]
@rmackay9 maybee you should open an issue to document your AP_Notify::flags.failsafe concerns. Otherwise they will get lost.

**Pull request closed without merging**